src/Microsoft.FeatureManagement/FeatureFilters/Recurrence/RecurrenceValidator.cs

```diff
@@ -341,49 +341,38 @@ private static bool TryValidateNumberOfOccurrences(TimeWindowFilterSettings sett
         private static bool IsDurationCompliantWithDaysOfWeek(TimeSpan duration, int interval, IEnumerable<DayOfWeek> daysOfWeek, DayOfWeek firstDayOfWeek)
         {
             Debug.Assert(interval > 0);
+            Debug.Assert(daysOfWeek.Count() > 0);
 
             if (daysOfWeek.Count() == 1)
             {
                 return true;
             }
 
-            DateTime firstDayOfThisWeek = DateTime.Today.AddDays(
-                DaysPerWeek - CalculateWeeklyDayOffset(DateTime.Today.DayOfWeek, firstDayOfWeek));
-
             List<DayOfWeek> sortedDaysOfWeek = SortDaysOfWeek(daysOfWeek, firstDayOfWeek);
 
-            DateTime prev = DateTime.MinValue;
+            DayOfWeek firstDay = sortedDaysOfWeek.First(); // the closest occurrence day to the first day of week
+
+            DayOfWeek prev = firstDay;
 
             TimeSpan minGap = TimeSpan.FromDays(DaysPerWeek);
 
-            foreach (DayOfWeek dayOfWeek in sortedDaysOfWeek)
+            foreach (DayOfWeek dayOfWeek in sortedDaysOfWeek.Skip(1))
```

Comment:
I don't like this Skip syntax- it seems less readable, could you add a comment why we skip the first? 

I'd prefer we don't do a skip and explicitly check `if prev == firstDay` or `if prev != null` instead.

Reply:
Change to 

``` C#
DayOfWeek firstDay = sortedDaysOfWeek.First(); // the closest occurrence day to the first day of week

DayOfWeek prev = firstDay;

TimeSpan minGap = TimeSpan.FromDays(DaysPerWeek);

foreach (DayOfWeek dayOfWeek in sortedDaysOfWeek)
{
    if (dayOfWeek != prev)
    {
        TimeSpan gap = TimeSpan.FromDays(CalculateWeeklyDayOffset(dayOfWeek, prev));

        if (gap < minGap)
        {
            minGap = gap;
        }

        prev = dayOfWeek;
    }
}
```